src/main/java/org/apache/sling/graphql/core/engine/ScriptedTypeResolverProvider.java

```diff
@@ -0,0 +1,91 @@
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.sling.graphql.core.engine;
+
+import org.apache.sling.api.resource.AbstractResource;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceMetadata;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.api.scripting.SlingScript;
+import org.apache.sling.api.servlets.ServletResolver;
+import org.apache.sling.graphql.api.SlingTypeResolver;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import javax.servlet.Servlet;
+
+/**
+ * Resolves a SlingTypeResolver name to a SlingScript if available.
+ */
+@Component(service = ScriptedTypeResolverProvider.class)
+public class ScriptedTypeResolverProvider {
```

Comment:
Do we really need a scripted type provider? I understand it exists here by analogy with the scripted SlingDataFetcher but I'm not sure about the use cases for it.

Reply:
Actually, I tried to follow the concept used for DataFetchers.

Reply:
I talked offline with @bdelacretaz and we don't think that a `ScriptedTypeResolverProvider` makes sense for the following reasons:
* it's hard to test by developers, requiring ITs
* it's kind of cumbersome to write one